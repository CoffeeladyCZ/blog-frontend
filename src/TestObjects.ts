export const articleDetail = Object.freeze({
  articleId: '4c37d8d2-7742-4d41-8d97-d0ffd50ab92c',
  comments: [],
  title: 'Test article',
  imageId: '1q37d8d2-2345-4d41-8d97-a9jhg50ab92c',
  content: 'Test content.',
  perex: 'Test perex',
  createdAt: '2023-09-03T10:13:18.337374',
  lastUpdatedAt: '2023-09-03T11:13:18.337374'
});

export const articleList = [
  {
    articleId: '4c37d8d2-7742-4d41-8d97-d0ffd50ab92c',
    title: 'Test title',
    perex: 'Test perex',
    imageId: '1q37d8d2-2345-4d41-8d97-a9jhg50ab92c',
    createdAt: '2023-09-03T10:13:18.337374',
    lastUpdatedAt: '2023-09-03T10:13:18.337374'
  },
  {
    articleId: '5d37d8d2-7742-4d41-8d97-d0ffd50ab92c',
    title: 'Test title dva',
    perex: 'Test perex dva',
    imageId: '2d37d8d2-2345-4d41-8d97-a9jhg50ab92c',
    createdAt: '2023-09-01T10:13:18.337374',
    lastUpdatedAt: '2023-09-01T10:13:18.337374'
  }
];
