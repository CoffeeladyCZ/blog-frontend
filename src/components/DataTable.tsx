import React from 'react';

import { DataGrid, GridColDef, GridValidRowModel } from '@mui/x-data-grid';
import { styled } from '@mui/material/styles';

import { ArticleType } from '../types/Articles';

import LinearLoading from './LinearLoading';

type DataTablePropsType = {
  headerColumns: GridColDef<GridValidRowModel>[];
  articles: ArticleType[];
  loading: boolean;
};

const StyledDataGrid = styled(DataGrid)`
  border: none;

  & .MuiDataGrid-columnHeaderTitle {
    font-weight: 600 !important;
`;

const DataTable: React.FC<DataTablePropsType> = ({
  headerColumns,
  articles,
  loading
}: DataTablePropsType) => {
  const getRowId = (articles: any) => articles.articleId;

  const CustomNoRowsOverlays = () => {
    return <div>No data found.</div>;
  };

  const CustomLoading = () => {
    return <LinearLoading loading={loading} />;
  };

  return (
    <StyledDataGrid
      data-testid="dataTable"
      getRowId={getRowId}
      rows={articles}
      columns={headerColumns}
      loading={loading}
      initialState={{
        pagination: {
          paginationModel: { page: 0, pageSize: 10 }
        },
        sorting: {
          sortModel: [{ field: 'lastUpdatedAt', sort: 'desc' }]
        }
      }}
      slots={{
        loadingOverlay: CustomLoading,
        noRowsOverlay: CustomNoRowsOverlays
      }}
      pageSizeOptions={[10, 25, 50]}
      checkboxSelection
    />
  );
};

export default DataTable;
