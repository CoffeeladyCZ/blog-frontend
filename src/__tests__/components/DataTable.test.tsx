import { render, screen } from '@testing-library/react';

import { articleList } from '../../TestObjects';

import DataTable from '../../components/DataTable';

describe('DataTable', () => {
  const columns = [
    { field: 'title', headerName: 'Article title', flex: 1 },
    { field: 'perex', headerName: 'Perex', flex: 2 },
    { field: 'author', headerName: 'Author', flex: 1 },
    {
      field: 'actions',
      headerName: 'Actions',
      flex: 1,
      renderCell: () => <></>
    }
  ];

  test('should snapshot with data', () => {
    const container = render(
      <DataTable headerColumns={columns} articles={articleList} loading={false} />
    );
    expect(container).toMatchSnapshot();
  });

  test('component is succesfully render', async () => {
    render(<DataTable headerColumns={columns} articles={articleList} loading={false} />);
    const dataTableElement = await screen.getByTestId('dataTable');

    expect(dataTableElement).toBeInTheDocument();
    expect(screen.getByText('Article title')).toBeInTheDocument();
    expect(screen.getByText('Perex')).toBeInTheDocument();
    expect(screen.getByText('Test title')).toBeInTheDocument();
  });
});
