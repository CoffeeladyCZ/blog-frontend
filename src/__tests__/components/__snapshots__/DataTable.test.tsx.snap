// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable should snapshot with data 1`] = `[Function]`;
